up-core-api/uprotocol/core/udiscovery/v3/udiscovery.proto

```diff
@@ -25,332 +24,97 @@ option java_multiple_files = true;
 // Enables generation of generic service attributes in C++
 option cc_generic_services = true;
 
-// Platform USE Discovery Service Interface
+// Client facing APIs to the uProtocol Discovery Service.
+// 
+// The uDiscovery service is used to store information about devices and services namely their
+// addresses (in URI format), and properties of topics. This API is used by clients (uEs) to be
+// about to access the information within the database. Clients talk to their local uDiscovery service
+// who, when is unable to find what it is looking for, queries the central database to find the information.
+// UDiscovery works in heiarchial mode where the local uDiscovery service is the first point of contact
+// and the domain level service (for information about devices within a domain), and then central service
+// is the last point of contact.
+//
+// NOTE: the internal uDiscovery communication protocol (for how data is replicated between the local, domain,
+// and central) is NOT covered in this interface as this is ONLY the client APIs.
 service uDiscovery {
   option (uprotocol.service_name) = "core.udiscovery"; // Service name
   option (uprotocol.service_version_major) = 3;
   option (uprotocol.service_version_minor) = 0;
   option (uprotocol.service_id) = 1;
 
-  // uDiscovery Node Change Notification that sends the Update message
-  option (uprotocol.notification_topic) = {
-    id: 0x8000,
-      name: "NodeChange",
-      message: "Notification"
-  };
 
-    
-
-  // Used by any uProtocol application or service to find service instances
-  // locations. The passed UUri contains valid UEntity, UResource, and UAuthority information
-  // for a query.
-  // What is returned is a list of UUris that match the query.
-  // If lookup fails, the response will be a UStatus with
-  // 2. code.NOT_FOUND: No matching UUris were found
-  // 3. code.INVALID_ARGUMENT: The passed UUri is invalid
-  // 4. code.PERMISSION_DENIED: The caller does not have permission to perform the query
-  rpc LookupUri(LookupUriRequest) returns (LookupUriResponse) {
+  // Find Services. 
+  //
+  // Discover t authorities, instances, and versions of services based on the passed URI.
+  // The only mandatory portion of the URI that must be passed is the ue_id, the rest can be discovered.
+  // When you pass the wildcard for authority_name, the search will be done up to the domain, in order to seach 
```

Comment:
this doesn't seem to be in line with the examples provided in the README.
In general, IMHO we should try not to duplicate the information in both this proto3 file and the README. My preference is to provide the detailed docs in the README and only add sparse comments here ...